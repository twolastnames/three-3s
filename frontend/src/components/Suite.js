import React from 'react';
import PropTypes from 'prop-types';

Suite.propTypes = {
  name: PropTypes.string.isRequired,
  extras: PropTypes.array,
};

export function Suite({ name, extras }) {
  let reactKey = 2;
  const extension = (extras || []).map((extra) => (
    <li key={reactKey++}>{extra}</li>
  ));
  return (
    <div>
      <ul>
        <li key="1">
          <a href={`/architect/suites/{id}`}>{name}</a>
        </li>
        {extension}
      </ul>
    </div>
  );
}
