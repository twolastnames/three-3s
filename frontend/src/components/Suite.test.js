import React from 'react';
import { shallow } from 'enzyme';
import { Suite } from './Suite';

describe('Suite', () => {
  describe('its rendering', () => {
    it('renders without extras', () => {
      const wrapper = shallow(<Suite key="1" name="my suite" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders with an extra', () => {
      const wrapper = shallow(
        <Suite key="1" name="my suite" extra={[<p key="2">An extra</p>]} />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('renders with extras', () => {
      const wrapper = shallow(
        <Suite
          key="1"
          name="my suite"
          extra={[
            <p key="1">An extra 1</p>,
            <p key="2">An extra 2</p>,
            <p key="3">An extra 3</p>,
          ]}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
