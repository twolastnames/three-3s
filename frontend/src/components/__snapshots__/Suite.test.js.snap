// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suite its rendering renders with an extra 1`] = `ShallowWrapper {}`;

exports[`Suite its rendering renders with extras 1`] = `ShallowWrapper {}`;

exports[`Suite its rendering renders without extras 1`] = `ShallowWrapper {}`;
